kafka-client/src/main/java/dev/responsive/kafka/internal/clients/AsyncStreamsKafkaClientSupplier.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.clients;
+
+import static dev.responsive.kafka.internal.config.ConfigUtils.eosEnabled;
+
+import dev.responsive.kafka.api.async.internals.AsyncThreadPoolRegistry;
+import java.util.Map;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.StreamsConfig;
+
+public class AsyncStreamsKafkaClientSupplier implements KafkaClientSupplier {
```

Comment:
this class is probably worth unit testing

Reply:
Since I ended up taking out the whole "create different clients based on eos" thing, it's pretty dead simple now. Can still add a unit test for it if you'd like but idk what's left that's worth testing